File: backend/src/main/java/com/chicago/labs/domain/PairingList.java
package com.chicago.labs.domain;

import org.springframework.data.annotation.PersistenceConstructor;

import java.time.LocalDateTime;
import java.util.List;

public class PairingList {
    private String internalId;
    private List<Pair> pairingList;
    private LocalDateTime timestamp;

    public PairingList() {
    }

    @PersistenceConstructor
    public PairingList(final String internalId, final List<Pair> pairingList, final LocalDateTime timestamp) {
        this.internalId = internalId;
        this.pairingList = pairingList;
        this.timestamp = timestamp;
    }

    public void setTimestamp(LocalDateTime timestamp) {
        this.timestamp = timestamp;
    }

    public List<Pair> getPairingList() {
        return pairingList;
    }

    public LocalDateTime getTimestamp() {
        return timestamp;
    }
}

File: backend/src/main/kotlin/com/chicago/labs/domain/PairingList.kt
package com.chicago.labs.domain

import org.bson.types.ObjectId
import org.springframework.data.annotation.Id
import org.springframework.data.annotation.PersistenceConstructor
import org.springframework.data.mongodb.core.mapping.Document
import java.util.*

@Document
data class PairingList
@PersistenceConstructor
constructor(
        @Id val id: ObjectId? = null,
        val internalId: String? = null,
        var pairingList: List<Pair>? = null,
        var timestamp: Date? = null)

